Gordon360/Services/YearTermTableService.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Models.CCT.Context;
+using Gordon360.Models.ViewModels;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Gordon360.Services
+{
+    public class YearTermTableService : IYearTermTableService
+    {
+        private readonly CCTContext _context;
+
+        public YearTermTableService(CCTContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<YearTermTable?> GetCurrentTermAsync()
+        {
+            var today = DateTime.Today;
+
+            return await _context.YearTermTable
+                .Where(t => t.TRM_BEGIN_DTE <= today &&
+                            (t.TRM_CDE == "FA" || t.TRM_CDE == "SP" || t.TRM_CDE == "SU"))
+                .OrderByDescending(t => t.TRM_BEGIN_DTE)
+                .FirstOrDefaultAsync();
```

Comment:
I don't think this is the most useful way of determining the "current term", because often times we want to specify a term as the current term based on something other than the term begin dates.

For that reason, ISG maintains a term configuration table that includes a row which defines the current term for many of the background database processes that shape 360's data. I've created the stored procedure `[CCT].[dbo].[GetCurrentTerm]` in both train and production that will retrieve that defined current term. I think that will be a more useful and reliable definition of the "current term" for 360's perspective, which will align with the rest of 360's dataset.